bundles/org.openhab.binding.sonos/src/test/java/org/openhab/binding/sonos/internal/SonosXMLParserTest.java

```diff
@@ -79,4 +79,43 @@ public void getRadioTimeFromXML() throws IOException {
             assertEquals("Paris, France", result.get(2));
         }
     }
+
+    @Test
+    public void getMetaDataFromXML() throws IOException {
+        InputStream resourceStream = getClass().getResourceAsStream("/MetaData.xml");
+        assertNotNull(resourceStream);
+        final String xml = new String(resourceStream.readAllBytes(), StandardCharsets.UTF_8);
+        SonosMetaData sonosMetaData = SonosXMLParser.getMetaDataFromXML(xml);
+        assertEquals("-1", sonosMetaData.getId());
+        assertEquals("-1", sonosMetaData.getParentId());
+        assertEquals("Turn Down for What - Single", sonosMetaData.getAlbum());
+        assertEquals("DJ Snake & Lil Jon", sonosMetaData.getCreator());
+        assertEquals("Turn Down for What", sonosMetaData.getTitle());
+        assertEquals("object.item.audioItem.musicTrack", sonosMetaData.getUpnpClass());
+        assertEquals("x-sonosapi-hls-static:librarytrack%3ai.eoD8VQ5SZOB8QX7?sid=204&flags=8232&sn=9",
+                sonosMetaData.getResource());
+        assertEquals(
+                "/getaa?s=1&u=x-sonosapi-hls-static%3alibrarytrack%253ai.eoD8VQ5SZOB8QX7%3fsid%3d204%26flags%3d8232%26sn%3d9",
+                sonosMetaData.getAlbumArtUri());
+    }
+
+    @Test
+    public void compileMetadataString() {
+        SonosEntry sonosEntry = new SonosEntry("1", "Can't Buy Me Love", "0", "A Hard Day's Night", "", "",
+                "&gt;object.item.audioItem.musicTrack&lt;", "");
```

Comment:
Why do you include "<" and ">" around the UPnP class ? It does not match something real.
In case it is just to test the correct escaping of XML characters, I would prefer (more realist) you put them in the title.

Reply:
I believe I found it somewhere else in the code, but tbh I'm not into the details. What would be something real to use here?

Reply:
Ah, copy'n'paste mistake when I found it here - included part of the XML tag:
https://github.com/openhab/openhab-addons/blob/84db35d656e4239126389eb05ff9829d5da3e2b4/bundles/org.openhab.binding.sonos/src/main/java/org/openhab/binding/sonos/internal/SonosXMLParser.java#L676C807-L676C807